src/timesteppers.jl

```diff
@@ -165,17 +187,29 @@ struct FilteredRK4TimeStepper{T,Tf} <: AbstractTimeStepper{T}
   filter::Tf
 end
 
-function RK4TimeStepper(eq)
+function RK4TimeStepper(eq::Equation)
```

Comment:
Should this constructor call the other constructor with `dev=CPU()` (rather than providing an independent implementation of the constructor...)?

Reply:
Ditto for all the other constructors.

Reply:
but will the first constructor always know dev?

Reply:
ok, reformed the code, e.g.,
```julia
function RK4TimeStepper(eq::Equation, dev::Device=CPU())
```